sale_custom_rounding/models/sale_order.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    tax_calculation_rounding_method = fields.Selection(
+        [
+            ("round_per_line", "Round per Line"),
+            ("round_globally", "Round Globally"),
+        ],
+        compute="_compute_tax_calculation_rounding_method",
+        store=True,
+        readonly=False,
+        help="How total tax amount is computed. If no value selected, "
+        "the method defined in the company is used.",
+    )
+
+    @api.depends("partner_invoice_id")
+    def _compute_tax_calculation_rounding_method(self):
+        for sel in self:
+            tax_calculation_rounding_method = False
+            if (
+                sel.partner_invoice_id
+                and sel.partner_invoice_id.tax_calculation_rounding_method
+            ):
+                tax_calculation_rounding_method = (
+                    sel.partner_invoice_id.tax_calculation_rounding_method
+                )
+            sel.tax_calculation_rounding_method = tax_calculation_rounding_method
+
+    @api.depends("tax_calculation_rounding_method")
+    def _compute_tax_totals_json(self):
+        for sel in self.filtered(lambda a: a.tax_calculation_rounding_method):
```

Comment:
call it record, not sel, it is confusing

Reply:
Also, I think you can simplify the code

you should probably do:

```
super(
            SaleOrder, self.filtered(lambda a: not a.tax_calculation_rounding_method)
        )._compute_tax_totals_json()
super(
            SaleOrder, self.filtered(lambda a: a.tax_calculation_rounding_method == "round_per_line").with_context
            (round=True
        )._compute_tax_totals_json()
super(
            SaleOrder, self.filtered(lambda a: a.tax_calculation_rounding_method == "round_globally").with_context
            (round=False
        )._compute_tax_totals_json()

```

Reply:
Done! Thank you.